index.js

```diff
@@ -2,21 +2,20 @@
 const ipRegex = require('ip-regex');
 const tlds = require('tlds');
 
-module.exports = options => {
-	options = {
-		strict: true,
-		...options
-	};
+module.exports = function (options) {
+	options = Object.assign({strict: true}, options);
 
-	const protocol = `(?:(?:[a-z]+:)?//)${options.strict ? '' : '?'}`;
-	const auth = '(?:\\S+(?::\\S*)?@)?';
-	const ip = ipRegex.v4().source;
-	const host = '(?:(?:[a-z\\u00a1-\\uffff0-9][-_]*)*[a-z\\u00a1-\\uffff0-9]+)';
-	const domain = '(?:\\.(?:[a-z\\u00a1-\\uffff0-9]-*)*[a-z\\u00a1-\\uffff0-9]+)*';
-	const tld = `(?:\\.${options.strict ? '(?:[a-z\\u00a1-\\uffff]{2,})' : `(?:${tlds.sort((a, b) => b.length - a.length).join('|')})`})\\.?`;
-	const port = '(?::\\d{2,5})?';
-	const path = '(?:[/?#][^\\s"]*)?';
-	const regex = `(?:${protocol}|www\\.)${auth}(?:localhost|${ip}|${host}${domain}${tld})${port}${path}`;
+	var protocol = `(?:(?:[a-z]+:)?//)${options.strict ? '' : '?'}`;
+	var auth = '(?:\\S+(?::\\S*)?@)?';
+	var ip = ipRegex.v4().source;
+	var host = '(?:(?:[a-z\\u00a1-\\uffff0-9][-_]*)*[a-z\\u00a1-\\uffff0-9]+)';
+	var domain = '(?:\\.(?:[a-z\\u00a1-\\uffff0-9]-*)*[a-z\\u00a1-\\uffff0-9]+)*';
+	var tld = `(?:\\.${options.strict ? '(?:[a-z\\u00a1-\\uffff]{2,})' : `(?:${tlds.sort(function (a, b) {
```

Comment:
Could be useful to list in the description the two changes that you make here:
1. `const` to `var`
2. remove arrow functions